src/body.rs

```diff
@@ -65,7 +65,7 @@ impl SizedReader {
 impl Payload {
     pub fn into_read(self) -> SizedReader {
         match self {
-            Payload::Empty => SizedReader::new(None, Box::new(empty())),
+            Payload::Empty => SizedReader::new(Some(0), Box::new(empty())),
```

Comment:
Is this change needed for the overall logic of this PR?

Reply:
I needed a way to differentiate `Payload::Reader` and `Payload::Empty`, `None` allowed me to express that the `SizedReader` has an unknown size, as opposed to `Some(0)` which express an empty Reader.